internal/server/instance/drivers/edk2/driver_edk2.go

```diff
@@ -178,16 +178,21 @@ func GetAchitectureFirmwarePairs(hostArch int) []FirmwarePair {
 // default installation path when constructing Code & Vars paths.
 func GetArchitectureFirmwarePairsForUsage(hostArch int, usage FirmwareUsage) []FirmwarePair {
 	firmwares := make([]FirmwarePair, 0)
+	incusEdk2Path := GetenvEdk2Path()
 
 	for _, installation := range GetArchitectureInstallations(hostArch) {
 		usage, found := installation.Usage[usage]
 		if found {
-			for _, firmwarePair := range usage {
-				searchPath := GetenvEdk2Path()
-				if searchPath == "" {
-					searchPath = installation.Path
-				}
+			searchPath := installation.Path
+			if !util.PathExists(searchPath) {
+				searchPath = incusEdk2Path
+			}
+
+			if searchPath == "" {
+				continue
+			}
```

Comment:
Should add a comment on top of that code to explain what's going on here, basically that since we have a user-specified firmware path, we'll check in there too.

Reply:
done